pkg/instancetype/instancetype.go

```diff
@@ -602,6 +619,15 @@ func applyMemory(field *k8sfield.Path, instancetypeSpec *instancetypev1alpha2.Vi
 		return Conflicts{field.Child("domain", "memory")}
 	}
 
+	hasMemoryGuest := false
+	if condition := instancetypeSpec.Memory.Guest; !condition.IsZero() {
```

Comment:
Please move these conflict conditions to separate function (something like `resolveMemoryConflicts`).